code/modules/projectiles/gun.dm

```diff
@@ -199,7 +199,25 @@
 		if(!(target && target.loc))
 			target = targloc
 			pointblank = 0
-
+	var/obj/projectile = consume_next_projectile(user)

```

Comment:
These change how a gun displays its fire message and its feedback. Old-code wise a gun would display fire message and feedback whenever a projectile was created. This new revision makes it display fire message only once, when the user actually fires the gun. Audio and visual feedback is kept to the projectile. This reduces chatlog spam greatly when burst-fire weapons are employed.